lib/bearerstrategy.js

```diff
@@ -264,101 +338,53 @@ Strategy.prototype.authenticate = function authenticateStrategy(req) {
    */
   async.waterfall([
 
-    // compute metadata url
+    // compute metadataUrl
     (next) => {
-      if (!self._options.certificate && !self._options.identityMetadata) {
-        var message = 'options.certificate and options.identityMetadata are both null. ' +
-            'It is not possible to validate a JWT token, BearerStrategy requires either ' +
-            'a PEM encoded public key or a metadata location that contains cert data ' +
-            ' for RSA and ECDSA callback';
-        log.warn(message);
-        return next(new TypeError(message));
-      }
-
-      // (1) no metadata but have a certificate, pass `null` to the next function, 
-      // which will skip the metadata loading
-      if (!self._options.identityMetadata)
-        return next(null, null);
-
-      // (2) there is only metadata, we calulate the metadataURL and pass it to the
-      // next function for metadata loading
-
-      // default key for metadata cache
-      var cacheKey = 'ordinary';
-
-      var metadataURL = self._options.identityMetadata;
-
-      if (self._options.policyName) {
-        log.info('B2C: using policy %s', self._options.policyName);
-        
-        if (!self._options.tenantName) {
-          return next(new TypeError('B2C: BearerStrategy requires you to pass the tenant name if using a B2C tenant.'));
-        }
+      params.metadataURL = self._options.identityMetadata
+            .concat(`?${aadutils.getLibraryProductParameterName()}=${aadutils.getLibraryProduct()}`)
+            .concat(`&${aadutils.getLibraryVersionParameterName()}=${aadutils.getLibraryVersion()}`); ;
 
-        // We are replacing the common endpoint with the concrete metadata of a B2C tenant.
-        metadataURL = metadataURL
-            .replace('common', self._options.tenantName)
-            .concat(`?p=${self._options.policyName}`)
-            .concat(`&${aadutils.getLibraryProductParameterName()}=${aadutils.getLibraryProduct()}`)
-            .concat(`&${aadutils.getLibraryVersionParameterName()}=${aadutils.getLibraryVersion()}`);
-
-        // use the policy name as the metadata cache key
-        cacheKey = 'policy: ' + policyName;
+      if (self._options.isB2C) {
+        params.cacheKey = 'policy: ' + self._options.policyName;
+        params.metadataURL = params.metadataURL.concat(`&p=${self._options.policyName}`)
       } else {
-        metadataURL = metadataURL.concat(`?${aadutils.getLibraryProductParameterName()}=${aadutils.getLibraryProduct()}`)
-        .concat(`&${aadutils.getLibraryVersionParameterName()}=${aadutils.getLibraryVersion()}`);
+        params.cacheKey = 'odinary';
```

Comment:
It can be anything we like. We used the same word 'ordinary' for non-B2C in OIDCStrategy too.